internal/crypto/keystores/keystore.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package keystores contains logic for loading encryption keys from a keystores
+package keystores
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+
+	serverconfig "github.com/stacklok/minder/internal/config/server"
+	"github.com/stacklok/minder/internal/crypto/algorithms"
+)
+
+//go:generate go run go.uber.org/mock/mockgen -package mock_$GOPACKAGE -destination=./mock/$GOFILE -source=./$GOFILE
+
+// KeyStore represents a struct which stores or can fetch encryption keys.
+type KeyStore interface {
+	// GetKey retrieves the key for the specified algorithm by key ID.
+	GetKey(algorithm algorithms.Type, id string) ([]byte, error)
+}
+
+// ErrUnknownKeyID is returned when the Key ID cannot be found by the keystore.
+var ErrUnknownKeyID = errors.New("unknown key id")
+
+// This structure is used by the keystore implementation to manage keys.
+type keysByAlgorithmAndID map[algorithms.Type]map[string][]byte
```

Comment:
Does this track the keys on disk only? I'm not sure I'm understanding the type key here, because the FS key (the KEK) is not tied to an algo, it's just random bytes.